### DISCUSSION TITLE
Features - How to separate them and implement them properly.

### DISCUSSION THREAD

**Discussion by wtrocki** [aerogear/charmil#57] [Ideas]
Once we add 1 or two features to the system we will need to improve overall visibility and UX of the library. 
What we want to avoid is just unusable implementation 

Currently we have very thin core with factory being the main usage and some examples for features that we adding. 
Let's discuss how we want to drive features for them to be available in one packaged scenario and how we could make them self documented.

What I see right now (in main) factory and it is interface is main driver for everything. 
if that is true we need to document this in getting started and focus on documenting everything around the factory.

Things to decide upon?

1. Do we actually need whole cobra commands for logging, i18n etc. - small micro examples could work? One uber example of single plugin

**Comment by ankithans** [comment]
> What I see right now (in main) factory and it is interface is main driver for everything.
if that is true we need to document this in getting started and focus on documenting everything around the factory.

yes factory is driving everything. All the features are served from factory to commands. #55 

> Do we actually need whole cobra commands for logging, i18n etc. - small micro examples could work? One uber example of single plugin

If I am right you are saying that pack all the features one example command. yes this would be easier to understand, rather than putting some feature in some command, other somewhere
